modules/internal/ChapelArray.chpl

```diff
@@ -4212,6 +4212,7 @@ module ChapelArray {
   }
 
   pragma "init copy fn"
+  pragma "unchecked throws"
```

Comment:
I don't think this function should use unchecked throws.
It should be apparent to the user when they are evaluating an iterator that throws.
E.g.

``` chapel
var A = for i in 1..100 do myThrowingFunction(i);
```

should need to be in a try/catch or in a function that throws.